onmt/translate/translation_server.py

```diff
@@ -290,16 +296,13 @@ def load(self):
         timer.tick("model_loading")
         if self.preprocess_opt is not None:
             self.logger.info("Loading preprocessor")
-            self.preprocessor = {}
-            if self.preprocess_opt['ZH_simplify'] is True:
-                from snownlp import SnowNLP
-                self.preprocessor['ZH_simplify'] = SnowNLP
-            if self.preprocess_opt['ZH_segmentation'] is True:
-                import pkuseg
-                seg = pkuseg.pkuseg()
-                self.preprocessor['ZH_segmentation'] = seg
+            self.preprocessor = []
+            for function_path in self.preprocess_opt:
+                function = get_function_by_path(function_path)
+                self.preprocessor.append(function)
+
             if len(self.preprocessor) == 0:
-                raise ValueError("Invalid value for preprocess.")
+                print("No preprocess function loaded.")
```

Comment:
Shouldn't we put a try/except around the for loop rather than printing a warning?